internal/service/customer.go

```diff
@@ -46,12 +46,64 @@ func (s *customerService) CreateCustomer(ctx context.Context, req dto.CreateCust
 			Mark(ierr.ErrValidation)
 	}
 
+	// Validate integration entity mappings if provided
+	if len(req.IntegrationEntityMapping) > 0 {
+		integrationService := NewIntegrationService(s.ServiceParams)
+		if err := integrationService.ValidateIntegrationEntityMappings(ctx, req.IntegrationEntityMapping); err != nil {
+			return nil, err
+		}
+	}
+
 	if err := s.DB.WithTx(ctx, func(txCtx context.Context) error {
 		if err := s.CustomerRepo.Create(txCtx, cust); err != nil {
 			// No need to wrap the error as the repository already returns properly formatted errors
 			return err
 		}
 
+		// Create integration entity mappings if provided
+		if len(req.IntegrationEntityMapping) > 0 {
```

Comment:
since this calls an external API better to not have this within a txn lets fix this in a follow up PR